morpheus/_lib/src/io/loaders/rest.cpp

```diff
@@ -17,21 +17,238 @@
+    py::object ep_pystr = py::str(ep);
+
+    // Following the syntax specifications in RFC 1808, urlparse recognizes a netloc only if it is properly introduced
+    // by ‘//’. Otherwise the input is presumed to be a relative URL and thus to start with a path component. 
+    // ref: https://docs.python.org/3/library/urllib.parse.html
+    // Workaround here is to prepend "//" if endpoint does not already include one
+    if (ep_pystr.attr("find")("//").cast<int>() == -1)
+    {
+        ep = "//" + ep;
+    }
+
+    py::object result  = urllib.attr("urlparse")(ep);
+    std::string host   = result.attr("hostname").is_none() ? "" : result.attr("hostname").cast<std::string>();
+    std::string target = result.attr("path").is_none() ? "" : result.attr("path").cast<std::string>();
+    if (target.empty())
+    {
+        target = "/";
+    }
+    std::string query = result.attr("query").is_none() ? "" : result.attr("query").cast<std::string>();
+    // if params exists, overrides query part included in endpoint
+    if (!params.empty())
+    {
+        query = params;
+    }
+    py::gil_scoped_release gil_release;
+
+    net::io_context ioc;
+    tcp::resolver resolver(ioc);
+    beast::tcp_stream stream(ioc);
+    try
+    {
+        auto const endpoint_results = resolver.resolve(host, PORT);
+        http::verb verb;
+        if (method == "GET")
+        {
+            verb = http::verb::get;
+        }
+        else if (method == "POST")
+        {
+            verb = http::verb::post;
+        }
+        http::request<http::string_body> request{verb, target + query, 11};
+        request.set(http::field::host, host);
+        request.set(http::field::user_agent, BOOST_BEAST_VERSION_STRING);
+        if (method == "POST")
+        {
+            request.set(http::field::content_type, content_type);
+            request.body() = body;
+            request.prepare_payload();
+        }
+        for (auto& x_header : x_headers)
+        {
+            request.insert(x_header.first, x_header.second);
+        }
+
+        int interval_milliseconds = 1000;
+        while (max_retry-- > 0)
+        {
+            stream.connect(endpoint_results);
+            http::write(stream, request);
+            beast::flat_buffer buffer;
+            http::read(stream, buffer, response);
+            beast::error_code ec;
+            stream.socket().shutdown(tcp::socket::shutdown_both, ec);
+            if (ec && ec != beast::errc::not_connected)
+            {
+                throw beast::system_error{ec};
+            }
+            int status = response.result_int();
+            // 503 Service Unavailable / 504 Gateway Timeout
+            if (status != 503 && status != 504)
+            {
+                break;
+            }
+            std::this_thread::sleep_for(std::chrono::milliseconds(interval_milliseconds));
+            interval_milliseconds *= 2;
+        }
+
+    } catch (std::exception const& e)
+    {
+        // TODO: need VLOG here?
```

Comment:
Yes, please. Good to avoid printing to any standard streams directly and let the logger handle it.